girder/models/file.py

```diff
@@ -93,23 +94,62 @@ def download(self, file, offset=0, headers=True, endByte=None,
         :type contentDisposition: str or None
         :type extraParameters: str or None
         """
+        if offset == 0:
+            events.trigger('model.file.download.before', info={'file': file})
+        else:
+            events.trigger('model.file.download.during', info={'file': file})
```

Comment:
Combine the before and during event into one? Or combine before, during, and after into one?